### PR TITLE
`Chore`: Add unit tests for architectural and AOP analysis

### PR DESCRIPTION
**Pull request by sarpsahinalp**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Ares
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

To improve the robustness and reliability of the project, we introduced new features such as additional unit tests and supporting classes, significantly enhancing test coverage and functionality validation. These updates include leveraging Mockito for advanced mocking capabilities and expanding Java architecture testing with rules for ClassLoaders and thread management. Bug fixes focused on improving parameter validation for secure testing. Additionally, documentation was updated to reflect recent advancements, including methods related to the Java Module System. Supporting these changes, new dependencies were added to strengthen testing frameworks, ensuring a more comprehensive and maintainable codebase.

- **New Features**
  - Introduced new unit tests for various classes, enhancing test coverage and validation of functionalities.
  - Added new classes such as `EmptyClass`, `ExampleClass`, and `CustomConditionEvents` to support functionality and testing.
  - Enhanced testing capabilities with the addition of Mockito for mocking behaviors in unit tests.
  - Expanded Java architecture testing with new rules and constants related to ClassLoaders and thread management.

- **Bug Fixes**
  - Improved constructor parameter validation in `JavaArchUnitSecurityTestCase`.

- **Documentation**
  - Updated reflection methods documentation to include new Java Module System methods.

- **Chores**
  - Added new dependencies to enhance testing frameworks.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Release Notes

- **New Features**
  - Enhanced testing capabilities with the addition of the Mockito framework.
  - New constructors and methods in the `JavaArchUnitSecurityTestCase` class for improved instantiation options.
  - Introduction of various new classes and methods to support testing of architecture and security policies.

- **Bug Fixes**
  - Removed obsolete test methods that were causing issues in the test suite.

- **Documentation**
  - Expanded documentation for reflection and thread manipulation methods to improve developer understanding.

- **Tests**
  - Comprehensive new test classes added for various components, ensuring robust coverage of functionality and security aspects.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,6 +57,12 @@
             <artifactId>junit-platform-launcher</artifactId>
             <version>${junit-platform-version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
+            <version>5.2.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (MarkusPaulsen) [inline pom.xml:65]: ToDo: Check if that isnecessary and delete if it is not

**Review by MarkusPaulsen — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (az108) [reply pom.xml:65]: necessary

**Review by MarkusPaulsen — APPROVED** [review]
Code looks good

**Review by az108 — APPROVED** [review]
Code LGTM

**Pull request merged**